icechunk-python/src/conflicts.rs

```diff
@@ -0,0 +1,203 @@
+            PyConflictType::UserAttributesUpdateOfDeletedNode => {
+                "User attributes update of deleted node"
+            }
+            PyConflictType::ChunkDoubleUpdate => "Chunk double update",
+            PyConflictType::ChunksUpdatedInDeletedArray => {
+                "Chunks updated in deleted array"
+            }
+            PyConflictType::ChunksUpdatedInUpdatedArray => {
+                "Chunks updated in updated array"
+            }
+            PyConflictType::DeleteOfUpdatedArray => "Delete of updated array",
+            PyConflictType::DeleteOfUpdatedGroup => "Delete of updated group",
+        };
+        write!(f, "{}", variant_str)
+    }
+}
+
+#[pymethods]
+impl PyConflictType {
+    fn __repr__(&self) -> String {
+        format!("{:?}", self)
+    }
+
+    fn __str__(&self) -> String {
+        format!("{}", self)
+    }
+}
+
+#[pyclass(name = "Conflict")]
+#[derive(Debug, Clone)]
+pub struct PyConflict {
+    #[pyo3(get)]
+    conflict_type: PyConflictType,
+    #[pyo3(get)]
+    path: String,
+}
+
+#[pymethods]
+impl PyConflict {
+    fn __repr__(&self) -> String {
+        format!("Conflict({:?}, path={})", self.conflict_type, self.path)
+    }
+
+    fn __str__(&self) -> String {
+        format!("{}: {}", self.path, self.conflict_type)
+    }
+}
+
+impl From<&Conflict> for PyConflict {
+    fn from(conflict: &Conflict) -> Self {
+        match conflict {
+            Conflict::NewNodeConflictsWithExistingNode(path) => PyConflict {
+                conflict_type: PyConflictType::NewNodeConflictsWithExistingNode,
+                path: path.to_string(),
+            },
+            Conflict::NewNodeInInvalidGroup(path) => PyConflict {
+                conflict_type: PyConflictType::NewNodeInInvalidGroup,
+                path: path.to_string(),
+            },
+            Conflict::ZarrMetadataDoubleUpdate(path) => PyConflict {
+                conflict_type: PyConflictType::ZarrMetadataDoubleUpdate,
+                path: path.to_string(),
+            },
+            Conflict::ZarrMetadataUpdateOfDeletedArray(path) => PyConflict {
+                conflict_type: PyConflictType::ZarrMetadataUpdateOfDeletedArray,
+                path: path.to_string(),
+            },
+            Conflict::UserAttributesDoubleUpdate { path, node_id: _ } => PyConflict {
+                conflict_type: PyConflictType::UserAttributesDoubleUpdate,
+                path: path.to_string(),
+            },
+            Conflict::UserAttributesUpdateOfDeletedNode(path) => PyConflict {
+                conflict_type: PyConflictType::UserAttributesUpdateOfDeletedNode,
+                path: path.to_string(),
+            },
+            Conflict::ChunkDoubleUpdate { path, node_id: _, chunk_coordinates: _ } => {
+                PyConflict {
+                    conflict_type: PyConflictType::ChunkDoubleUpdate,
+                    path: path.to_string(),
```

Comment:
@mpiannucci Can we create a ticket to add the chunk_coordinates to the exception? Without the coordinates users will have a hard time understanding why they have a conflict in their algorithms. It doesn't need to happen in this PR.

Reply:
Isnt the path the same as the coordinates? I will confirm..

Reply:
Ahhhh it is not. Sounds good that is important